### DISCUSSION TITLE
Bun.build - what about CSS?

### DISCUSSION THREAD

**Discussion by futured-it** [oven-sh/bun#7119] [Q&A] [answered]
Hey guys! Well, I tried to use Bun.build, and, after ESBuild (which Bun.build authors inspired from), I have problems with CSS. I have a React app, where I have different CSS files imported in JSX. When I'm building with ESBuild, I have one compiled CSS file, and Bun.build at the same time just puts all CSS's as assets (in different files).

The question is - I'm doing something wrong? or Bun.build doesn't support CSS building right now?

**Comment by abuisman** [comment]
I had the same issue. I tried implementing bun with our Rails app for a day and didn't get anywhere close. Then within 10 minutes I had the build working with esbuild. It was a shame, because I'd have preferred to go with bun.

**Reply by futured-it** [reply, marked as answer]
Yeah, well, Bun.build isn't production ready, and doesn't work with CSS. But you can use an external CSS compiler like Lightning CSS. Someone even wrote a plugin for Bun to use it. (https://github.com/wobsoriano/bun-lightningcss)

Well, right now, until you have a really massive codebase, you can barely see any performance difference between Bun and ESBuild. But technically, both Bun JS compiler and Lightning CSS compiler are faster than ESBuild ones.

**Reply by abuisman** [reply]
Good to know thanks. 

Yeah it is fine now with esbuild. I can't imagine spending any time on it in the next 2-3 years to switch to bun, but it would've been nice since I like the general idea of bun.
